pkg/internal/http/backoff.go

```diff
@@ -0,0 +1,67 @@
+package http
+
+import (
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+)
+
+// Backoff holds configuration for the backoff mechanism for http requests
+type Backoff struct {
+	ready         func() bool   // Function to determine when we are done
+	wait          time.Duration // How much time to wait between retries
+	timelimit     time.Duration // Maximum time to wait between retries
+	withTimeLimit bool          // Whether we have a time limit
+}
```

Comment:
I found this interesting, I thought this would be better an interface, but I like how you implemented these.